tests/Gaufrette/Functional/Adapter/GoogleCloudClientStorageTest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Gaufrette\Functional\Adapter;
+
+use Gaufrette\Adapter\GoogleCloudClientStorage;
+use Gaufrette\Exception\FileNotFound;
+use Gaufrette\Filesystem;
+use Google\Cloud\Storage\Acl;
+use Google\Cloud\Storage\StorageClient;
+
+/**
+ * Functional tests for the GoogleCloudClientStorage adapter.
+ * Edit the phpunit.xml.dist Google Cloud Client adapter section for configuration
+ *
+ * @author  Lech Buszczynski <lecho@phatcat.eu>
+ */
+class GoogleCloudClientStorageTest extends FunctionalTestCase
+{
+    private $string    = 'Yeah mate. No worries, I uploaded just fine. Meow!';
+    private $directory = 'tests';
+    private $bucketName;
+    private $sdkOptions;
+    private $bucketOptions;
+
+    public function setUp()
+    {
+        $gccsProjectId   = getenv('GCCS_PROJECT_ID');
```

Comment:
It's strange to call these variables GCCS_* and it's actually strange to call this adapter `GoogleCloudClientStorage`. I mean, there's no need for that `Client`.